crates/ruff_python_ast/src/all.rs

```diff
@@ -14,29 +10,28 @@ bitflags! {
 }
 
 /// Extract the names bound to a given __all__ assignment.
-pub fn extract_all_names(
-    ctx: &Context,
-    stmt: &Stmt,
-    scope: &Scope,
-) -> (Vec<String>, AllNamesFlags) {
-    fn add_to_names(names: &mut Vec<String>, elts: &[Expr], flags: &mut AllNamesFlags) {
+pub fn extract_all_names<F>(stmt: &Stmt, is_builtin: F) -> (Vec<&str>, AllNamesFlags)
```

Comment:
Is there any downside to using a closures here (as opposed to passing in `Context`)? I'd like to do this in a few other places to remove dependencies on `Context`...

Reply:
Not necessarily. It can make it more difficult to call the function because I have to figure out what to pass for `is_builtin` whereas I probably know from where I can get `context`. The way I feel about it is that it's similar to using unnamed types. Using a function falls back to unnamed types because we don't have better abstractions. This can be a feasible choice but it convey's less information, making it more difficult to understand. 

Can we add some documentation for `is_builtin` to document its semantics?